### ISSUE TITLE
[Action] Make Grafana dashboards public accessible from cluster

### ISSUE THREAD

**Issue by rossf7** [cncf-tags/green-reviews-tooling#31]
This issue is to make our Grafana dashboards public accessible from our Equinix cluster.

We already have the Kepler dashboard deployed by Flux https://github.com/cncf-tags/green-reviews-tooling/issues/16

For https://github.com/cncf-tags/green-reviews-tooling/issues/20 we want to be able to visualize the metrics we are interested in collecting.

## Networking

- Currently the Equinix cluster managed by opentofu doesn't have a publicly accessible Ingress or Gateway API
- We can configure elastic IPs for our Equinix nodes via the TF provider https://deploy.equinix.com/developers/docs/metal/networking/elastic-ips/

## Open Questions

- Do we use Ingress? K3s can be configured to install Traefik https://docs.k3s.io/networking#traefik-ingress-controller
- Do we use Gateway API? Cilium has Gateway API support https://docs.cilium.io/en/stable/network/servicemesh/gateway-api/gateway-api/
- Do we need a publicly accessible DNS entry? e.g. tag-env-green-reviews.cncf.io
- Do we need to have a cert provisioned using cert-manager?

**Comment by dipankardas011** [comment]
I Can help with this issue

**Comment by rossf7** [comment]
@dipankardas011 thanks for taking #30 

@greenscale-nandesh thanks for taking this one. As of #36 the K3s install of Traefik is now deployed to the cluster.

As a starting point it would be worth investigating if we can enable the Grafana ingress via the [kube-prometheus-stack](https://github.com/cncf-tags/green-reviews-tooling/blob/4988fa1bdc2cf46acf3e9054ff517a144bd11897/clusters/base/kube-prometheus-stack.yaml) helm chart

https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack

We can then look into getting a DNS entry and cert. We should also investigate adding an elastic IP to the worker node.
https://deploy.equinix.com/developers/docs/metal/networking/elastic-ips/

Nandesh, let me know if anything is unclear or you want to sync on this.

**Comment by leonardpahlke** [comment]
Yes. I can open a ticket once we have an elastic IP and a dashboard to display smth. I would ask for `green-reviews.tag-env-sustainability.cncf.io`

**Comment by rossf7** [comment]
@nikimanoledaki @leonardpahlke @AntonioDiTuri  Grafana is now available via the elastic ip '147.75.40.83'  should we go with `green-reviews.tag-env-sustainability.cncf.io` as the domain? 

You can access it at http://147.75.40.83/

**Comment by dipankardas011** [comment]
Given: docs/home page `tag-env-sustainability.cncf.io`

we have this subdomain `*.green-reviews.tag-env-sustainability.cncf.io`

**Comment by dipankardas011** [comment]
for what all things we need to decide the domain for?

**Comment by leonardpahlke** [comment]
We can get the domain registered. If we plan to have other pages too we could also opt for `dash-green-reviews.tag-env-sustainability.cncf.io` or `dash.green-reviews.tag-env-sustainability.cncf.io` & later `docs.greeen-reviews.tag-env-sustainability.cncf.io`.

**Comment by AntonioDiTuri** [comment]
+1 For 
- dash.green-reviews.tag-env-sustainability.cncf.io
- docs.greeen-reviews.tag-env-sustainability.cncf.io

**Comment by rossf7** [comment]
I think the dashboard is the primary site so what about using `green-reviews.tag-env-sustainability.cncf.io` for that?

+1 for docs.green-reviews.tag-env-sustainability.cncf.io

**Comment by dipankardas011** [comment]
> I think the dashboard is the primary site so what about using green-reviews.tag-env-sustainability.cncf.io for that?

what about `grafana.green-reviews.tag-env-sustainability.cncf.io` for the dasbooard
😄

**Comment by dipankardas011** [comment]
for the docs it might be good if we keep it as the root of the subdomain and all other in the subdomain like `*.green-reviews...`

**Comment by rossf7** [comment]
@leonardpahlke like we agreed in WG meeting domain will be `dash.green-reviews.tag-env-sustainability.cncf.io` and the elastic IP is `147.75.40.83`

**Comment by leonardpahlke** [comment]
👍 - `147.75.40.83` leads to a Grafana page where you need to provide credentials. Do we plan to make this page public?

**Comment by leonardpahlke** [comment]
will wait for https://github.com/cncf-tags/green-reviews-tooling/issues/65#issuecomment-2049163754 - so we register both CNAME’s at once

**Comment by dipankardas011** [comment]
had one question what are we going to use nginx lb or nginx gateway api?

**Comment by rossf7** [comment]
> 147.75.40.83 leads to a Grafana page where you need to provide credentials. Do we plan to make this page public?

@leonardpahlke We can share the SCI dashboard publicly but it will need code changes because we use template variables which is not supported https://github.com/grafana/grafana/issues/67346#issuecomment-1864389382

> had one question what are we going to use nginx lb or nginx gateway api?

@dipankardas011 Grafana already has an ingress managed via the helm chart. We use the traefik ingress controller that ships with k3s. 

We will likely need to add cert-manager to the cluster managed by Flux and configure LetsEncrypt to issue a cert. Once the domain is set up we can add that.
